docs/content/7.v7/12.lang-switcher.md

```diff
@@ -0,0 +1,157 @@
+# Lang Switcher
+
+When **@nuxtjs/i18n** loads in your app, it adds your `locales` configuration to `this.$i18n` (or `app.i18n`), which makes it really easy to display a lang switcher anywhere in your app.
+
+Here's an example lang switcher where a `name` key has been added to each locale object in order to display friendlier titles for each link:
+
+```vue
+<nuxt-link
+  v-for="locale in availableLocales"
+  :key="locale.code"
+  :to="switchLocalePath(locale.code)">{{ locale.name }}</nuxt-link>
+```
+
+```js
+computed: {
+  availableLocales () {
+    return this.$i18n.locales.filter(i => i.code !== this.$i18n.locale)
+  }
+}
+```
+
+```js {}[nuxt.config.js]
+i18n: {
+  locales: [
+    {
+      code: 'en',
+      name: 'English'
+    },
+    {
+      code: 'es',
+      name: 'Español'
+    },
+    {
+      code: 'fr',
+      name: 'Français'
+    }
+  ]
+}
+```
+
+<alert type="info">
+
+When using `detectBrowserLanguage` and wanting to persist locale on a route change, you must call one of the functions that update the stored locale cookie. Call either [`setLocaleCookie(locale)`](/api#setlocalecookie) to persist just the cookie locale or [`setLocale(locale)`](/api#setlocale) to both persist the cookie locale and switch the route to the specified locale. Otherwise, locale might switch back to the saved one during navigation.
```

Comment:
```suggestion
When using `detectBrowserLanguage` and wanting to persist locale on a route change, you must call one of the functions that update the stored locale cookie. Call either [`setLocaleCookie(locale)`](./api#setlocalecookie) to persist just the cookie locale or [`setLocale(locale)`](./api#setlocale) to both persist the cookie locale and switch the route to the specified locale. Otherwise, locale might switch back to the saved one during navigation.
```